compactor/src/compact.rs

```diff
@@ -353,6 +353,12 @@ impl Compactor {
             },
         );
 
+        let candidate_num = candidates.len();
+        debug!(
+            candidate_num,
+            "Number of candidate partitions to be considered to compact"
+        );
+
```

Comment:
I don't think it makes any difference, but you can do something like this for less code (and only invoke `candidates.len()` if debug level logging is enabled)

```suggestion
        debug!(
            candidate_num=candidates.len();,
            "Number of candidate partitions to be considered to compact"
        );

```